build.sbt

```diff
@@ -9,6 +11,19 @@ lazy val root = project.
 
 scalafmtOnCompile in ThisBuild := true
 
+// see https://www.wartremover.org/doc/warts.html for the full list.
+wartremoverErrors in (Compile, compile) ++= Seq(
```

Comment:
This isn't the full set of safe warts they recommend checking for, but many of the other ones were too intense to do as a small exercise. It's easy to add more to this list if we want to lock it down even tighter.